stdlib/LinearAlgebra/src/diagonal.jl

```diff
@@ -219,10 +219,32 @@ end
 (*)(D::Diagonal, B::AbstractTriangular) =
```

Comment:
broadcast based `mul!` also works for `AbstractTriangular`.
optimize this two in this PR as well?